src/hstr_utils.c

```diff
@@ -217,7 +217,7 @@ char *get_shell_name_by_ppid(const int pid)
           shell=strrchr(shell,'/');
           if(shell != NULL) {
               shell++;
-              strncpy(name,shell,sizeof(char)*strlen(shell));
+              strncpy(name,shell,sizeof(char)*PID_BUFFER_SIZE);
```

Comment:
I'm not sure about this change. I would be accessing memory which does not belong to `shell` variable and copying it to `name`...

Reply:
@dvorka How so?

>  The strncpy() function is similar, except that at most n bytes of src are copied.

If the max number of bytes isn't tied to `src`, then deriving it from`dst` serves no purpose. This change is correct, however lacks a reallocation of `name` which likely needs to be bigger than the initial `PID_BUFFER_SIZE` in order to fit `shell`. I opened #411 before noticing this PR, but #411 also handles the reallocation.